eth/gasprice/gasprice.go

```
@@ -260,25 +236,21 @@ func (oracle *Oracle) estimateNextBaseFee(ctx context.Context) (*big.Int, error)
 // SuggestPrice returns an estimated price for legacy transactions.
 func (oracle *Oracle) SuggestPrice(ctx context.Context) (*big.Int, error) {
 	// Estimate the effective tip based on recent blocks.
-	tip, baseFee, err := oracle.suggestDynamicFees(ctx)
+	tip, err := oracle.suggestTip(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	// We calculate the [nextBaseFee] if a block were to be produced immediately.
-	// If [nextBaseFee] is lower than the estimate from sampling, then we return it
-	// to prevent returning an incorrectly high fee when the network is quiescent.
+	// We calculate the `nextBaseFee` if a block were to be produced immediately.
 	nextBaseFee, err := oracle.estimateNextBaseFee(ctx)
-	if err != nil {
+	if nextBaseFee == nil {
+		// This occurs if AP3 has not been activated
+		return tip, nil
+	} else if err != nil {
```

Sorry for the back&forth, but I think it is unexpected to check `nextBaseFee == nil` _before_ `err != nil`. If `err != nil`, I don't think we should consider `nextBaseFee` a meaningful value here.

---

If `err != nil` for any reason, it isn't used. However, `nextBaseFee == nil` implies `err != nil` , but not vice-versa. I guess the real point here is what's the behavior pre-AP3? Prior to this PR, we would default to `suggestDynamicFees`, but that should not be returned. I don't think we should return an error on expected behavior.

---

internal implications should not matter for err handling in call places. Errs should (almost) always checked before the value is being used/check. 

I think in this case we can check err first and return the err(to keep it same behaviour with  `EstimateBaseFee`) and then perform addition if nextBaseFee is not nil.

---

the code got updated before my comment here.